extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,57 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(function waitTillReady() {
+      // Wait until the image has completed.
+      if (!imgPlaceholder.complete) {
+        return timer.promise(10).then(waitTillReady);
```

Comment:
I can do some more debugging to understand why it's not set. But one thing I found is that using `loadPromise` in `layoutCallback` sets `complete` correctly, however in `buildCallback` it doesn't, does that give an idea of what could be happening? 

Otherwise, I can work on adding `timer.pollUntil` and use that for now. Thanks for the code snippets :+1: